File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        Arrays.sort(nums);
        int count=0,left=0,right=nums.length-1;
        
        while(right>left){
            if(nums[left]+nums[right]>k)               
                right--;
            else if(nums[left]+nums[right]<k) 
                left++;
            else {
                count++;
                right--;
                left++;
            }
        }
        return count;
    }
}

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        beg,end = 0,len(nums)-1
        res = 0
        while end>beg:
            total = nums[end]+nums[beg]
            if total>k:
                end -= 1
            elif total<k:
                beg += 1
            else:
                end -= 1
                beg += 1
                res += 1
        return res